mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxLifecycleAwareObserver.kt

```diff
@@ -105,7 +116,12 @@ internal class MvRxLifecycleAwareObserver<T>(
             return
         }
         if (!isDisposed) {
-            val valueToDeliverOnUnlock = if (alwaysDeliverLastValueWhenUnlocked && lastValue != null) lastValue else lastUndeliveredValue
+            val valueToDeliverOnUnlock = when {
+                deliveryMode is UniqueOnly -> lastUndeliveredValue
+                deliveryMode is Standard && lastUndeliveredValue != null -> lastUndeliveredValue
+                deliveryMode is Standard && lastUndeliveredValue == null -> lastValue
```

Comment:
We always wrap properties in tuples, so they can never be null. I updated the generics to reflect this (RxJava doesn't allow null anyway)

Reply:
ah, great. In that case it seems the `deliveredFirstValue` atomic boolean is unnecessary because it will always be true when `lastDeliveredValueFromPriorObserver` is non null (I think), so checking both seems redundant.

it is maybe more clear though